From 27106db37ef586d7a3a214f20e7ffde359f67c2a Mon Sep 17 00:00:00 2001
From: John Zhuge <jzhuge@cloudera.com>
Date: Thu, 11 Aug 2016 13:36:06 -0700
Subject: [PATCH 1716/2848] HDFS-7934. Update RollingUpgrade rollback
 documentation: should use bootstrapstandby for
 standby NN. Contributed by J. Andreina.

(cherry picked from commit b172d03595d1591e7f542791224607d8c5fce3e2)
(cherry picked from commit 725621d95a833f813e2164e32b33984e1c1ee29a)
(cherry picked from commit e1ce0700eb2c5bf2b765e6a93df8146207a6fa9e)

Change-Id: I562a9770938c7b273ecb40e89c5a10aee76a85dd
---
 .../src/site/xdoc/HdfsRollingUpgrade.xml           |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/site/xdoc/HdfsRollingUpgrade.xml b/hadoop-hdfs-project/hadoop-hdfs/src/site/xdoc/HdfsRollingUpgrade.xml
index 1053695..85714b4 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/site/xdoc/HdfsRollingUpgrade.xml
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/site/xdoc/HdfsRollingUpgrade.xml
@@ -252,14 +252,16 @@
   <p>
     Rollback from a newer release to the pre-upgrade release is always supported.
     However, it cannot be done in a rolling fashion.  It requires cluster downtime.
+    Suppose <em>NN1</em> and <em>NN2</em> are respectively in active and standby states.
     Below are the steps for rollback:
   </p>
   <ul>
     <li>Rollback HDFS<ol>
       <li>Shutdown all <em>NNs</em> and <em>DNs</em>.</li>
       <li>Restore the pre-upgrade release in all machines.</li>
-      <li>Start <em>NNs</em> with the
+      <li>Start <em>NN1</em> as Active with the
         "<a href="#namenode_-rollingUpgrade"><code>-rollingUpgrade rollback</code></a>" option.</li>
+      <li>Run `-bootstrapStandby' on NN2 and start it normally as standby.</li>
       <li>Start <em>DNs</em> with the "<code>-rollback</code>" option.</li>
     </ol></li>
   </ul>
-- 
1.7.9.5

